src/PixiEditor/Views/Main/ViewportControls/Viewport.axaml

```diff
@@ -133,6 +134,26 @@
                                           IsChecked="{Binding SnappingEnabled, RelativeSource={RelativeSource Mode=FindAncestor, AncestorType=viewportControls:Viewport}, Mode=TwoWay}" />
                         </StackPanel>
                         <Separator />
+                        <StackPanel Spacing="10" HorizontalAlignment="Center" Orientation="Horizontal">
+                            <TextBlock VerticalAlignment="Center"
+                                       ui:Translator.Key="GRIDLINES_XSIZE" />
+                            <input:NumberInput Width="32"
+                                         HorizontalAlignment="Left"
+                                         Min="1" Decimals="0"
+                                         Max="1024"
+                                         Value="{Binding GridLinesXSize, RelativeSource={RelativeSource Mode=FindAncestor, AncestorType=viewportControls:Viewport}, Mode=TwoWay}" />
+                        </StackPanel>
+                        <Separator />
+                        <StackPanel Spacing="10" HorizontalAlignment="Center" Orientation="Horizontal">
+                            <TextBlock VerticalAlignment="Center"
+                                       ui:Translator.Key="GRIDLINES_YSIZE" />
+                            <input:NumberInput Width="32"
+                                         HorizontalAlignment="Left"
+                                         Min="1" Decimals="0"
+                                         Max="1024"
+                                         Value="{Binding GridLinesYSize, RelativeSource={RelativeSource Mode=FindAncestor, AncestorType=viewportControls:Viewport}, Mode=TwoWay}" />
+                        </StackPanel>
+                        <Separator />
```

Comment:
There shouldn't be a separator between Grid X size and Grid Y size in UI, I'd actually do it similarly to render preview option, so label on top and two inputs below like:
```
________________
    Grid Size: 
[    ]  [    ] 
-----------------
```
I believe there is no need to put X: before first input and Y: before second since it's pretty obvious, but it depends on how it will look.

Additionally, the inputs should be disabled if grid lines are disabled.

Reply:
I would also enable decimals since vectors are not confined to pixel grid

Reply:
![image](https://github.com/user-attachments/assets/090d5020-e294-4bf5-a6ac-c40b6e7fbe85)

How's this?

I'm not sure one way or the other about disabling the inputs when the grid is disabled. I personally tend to get annoyed when I can't configure the numbers inside of settings that are disabled.

Also, I've disabled decimals, but now it's non-obvious that the easiest way to get an integer number of pixels with the mouse is by using the scroll wheel (instead of clicking and dragging). Does the UI system have some way to make it use integers when using the mouse, but allow typing in decimals?

Reply:
Yeah, the UI looks better now, however now that I think about is, it might be too small for bigger numbers and will overflow? 

About disabling inputs; fair enough, if it's frustrating for you I guess there are more people who'd feel like that, you can leave it as it is.

There is no rounding/step option for mouse dragging, but feel free to add it.

Reply:
![image](https://github.com/user-attachments/assets/924ee13b-e0ab-4cd0-8e49-703097d03fb6)

How's this?

Reply:
I guess this solution is the most practical. You can also play around using `SizeInput` instead of `NumberInput` it has [px] after the text, it might fit better, but it's up to you. It's good enough now